modules/core/src/test/java/org/apache/ignite/internal/processors/cache/binary/GridCacheBinaryObjectsAbstractSelfTest.java

```diff
@@ -423,30 +424,77 @@ public void testMap() throws Exception {
     }
 
     /**
+     * Checks deserialization of elements in the singleton map.
+     *
+     * @throws Exception If failed.
+     */
+    public void testSingletonMap(){
+        IgniteCache<Integer, Map<TestObject,TestObject>> c = jcache(0);
+
+        TestObject obj = new TestObject(123);
+        c.put(0, Collections.singletonMap(obj,obj));
+
+        checkSingleElement(c.get(0).entrySet(), entry -> {
+            assertEquals(123, entry.getKey().val);
+            assertEquals(123, entry.getValue().val);
+        });
+
+        IgniteCache<Integer, Map<BinaryObject,BinaryObject>> kpc = keepBinaryCache();
+
+        Map<?,?> cBinary = kpc.get(0);
+
+        assertEquals(Collections.singletonMap(null,null).getClass(), cBinary.getClass());
+
+        checkSingleElement(cBinary.entrySet(), entry -> {
+            assertTrue(entry.getKey() instanceof BinaryObject);
+            assertTrue(entry.getValue() instanceof BinaryObject);
+            assertEquals(Integer.valueOf(123), ((BinaryObject)entry.getKey()).field("val"));
+            assertEquals(Integer.valueOf(123), ((BinaryObject)entry.getValue()).field("val"));
+        });
+    }
+
+    /**
+     * Checks deserialization of elements in the singleton list and set.
+     *
      * @throws Exception If failed.
      */
-    public void testSingletonList() throws Exception {
+    public void testSingletonCollection(){
```

Comment:
space after () and before {

Please install code style from ignite\idea\ignite_codeStyle.xml and run code reformat before commit

Reply:
Fixed

Reply:
Fixed